inginious/frontend/plugins/multilang/static/hdlgrader.js

```diff
@@ -0,0 +1,458 @@
+function updateWaveDromBlock(blockId) {
+    let block = $("#" + blockId);
+    let htmlbeg = block.html();
+    block.html(parseOutputDiff(htmlbeg)+parseHDL(htmlbeg));
+}
+
+function parseHDL(diff){
+  let lines = diff.split('\\n');
+  let inputsG = {};
+  let outputsG = {};
+  let inputsC = {};
+  let outputsC = {};
+
+  let noBinary = {};
+  let dataG = {};
+  let dataC = {};
+
+  let result = [];
+  let lastTimeGood = 0;
+  let lastTimeBad = 0;
+  result.push('<script src="http://wavedrom.com/skins/default.js" type="text/javascript"></script>');
+  result.push('<script src="http://wavedrom.com/wavedrom.min.js" type="text/javascript"></script>');
```

Comment:
Did you try loading this scripts as the other scripts are loaded in the [ __init__.py ](https://github.com/JuezUN/INGInious/blob/master/inginious/frontend/plugins/multilang/__init__.py#L28) file?

Reply:
Now it is loaded in the init file, if use_wavedrom is set to True in the configuration.yaml file